File: scripts/send_email.py
import os
import datetime
import smtplib
import subprocess
from dotenv import load_dotenv
from email import encoders
from email.mime.base import MIMEBase
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText

load_dotenv()

sender_email = os.getenv("EMAIL_USERNAME")
sender_password = os.getenv("EMAIL_PASSWORD")
recipient_email = os.getenv("EMAIL_USERNAME")

datetime = datetime.datetime.now().strftime('%d-%m-%Y-%H-%M-%S')

subject = f"TKDJ backup - {datetime}"

# Execute the Bash script and capture the machine
output = subprocess.check_output(['./get_stats.sh'])
body = output.decode() # include stats into email body

with open("backup.json", "rb") as attachment:
    # Add the attachment to the message
    part = MIMEBase("application", "octet-stream")
    part.set_payload((attachment).read())
encoders.encode_base64(part)
part.add_header(
    "Content-Disposition",
    f"attachment; filename={os.path.basename(f'backup-{datetime}.json')}"
)

message = MIMEMultipart()
message['Subject'] = subject
message['From'] = sender_email
message['To'] = recipient_email
html_part = MIMEText(body)
message.attach(html_part)
message.attach(part)

server = smtplib.SMTP_SSL('smtp.gmail.com', 465)
server.login(sender_email, sender_password)
server.sendmail(sender_email, recipient_email, message.as_string())
server.quit()

File: scripts/send_email.js
const nodemailer = require("nodemailer");
const { execSync } = require("child_process");
const fs = require("fs");
require('dotenv').config({
    path: `${__dirname}/../.env`
});

// Gmail account credentials
const user = process.env.EMAIL_USERNAME;
const pass = process.env.EMAIL_PASSWORD;

// Recipient email address
const to = process.env.EMAIL_USERNAME;

// Current date and time
const date = new Date().toLocaleString();

// Output of shell script
const output = execSync("./scripts/get_stats.sh").toString();

// File to attach
const file = "./backup.json";

// Create reusable transporter object using SMTP transport
const transporter = nodemailer.createTransport({
  service: "gmail",
  auth: {
    user: user,
    pass: pass
  }
});

// Read the contents of the file to attach
const attachment = {
  filename: "backup.json",
  content: fs.readFileSync(file)
};

// Email message
const message = {
  from: user,
  to: to,
  subject: `TKDJ: backup for ${date}`,
  text: output,
  attachments: [attachment]
};

// Send the email
transporter.sendMail(message, (err, info) => {
  if (err) {
    console.log(err);
  } else {
    console.log(`Email sent: ${info.response}`);
  }
});
